src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -65,6 +65,7 @@ internal static void Write(PackageSpec packageSpec, IObjectWriter writer, bool h
             SetDictionaryValues(writer, "scripts", packageSpec.Scripts);
 #pragma warning restore CS0612 // Type or member is obsolete
 
+            SetValue(writer, "SDKAnalysisLevel", packageSpec.SDKAnalysisLevel);
```

Comment:
This adds `SDKAnalysisLevel` to the dgspec file that will be used by the restore command